mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMailboxManager.java

```diff
@@ -488,8 +488,8 @@ public Mailbox deleteMailbox(MailboxId mailboxId, MailboxSession session) throws
                 Subscription subscription = new Subscription(session.getUser(), renamedResult.getOriginPath().getName());
                 if (subscriptionsForUser.contains(subscription)) {
                     subscriptionMapper.delete(subscription);
-                    subscriptionMapper.save(new Subscription(session.getUser(), renamedResult.getDestinationPath().getName()));
                 }
+                subscriptionMapper.save(new Subscription(session.getUser(), renamedResult.getDestinationPath().getName()));
```

Comment:
Why?

Reply:
We should have tests breaking here!

Reply:
Maybe my fault to make it break after rebase.